docs/source/guides/speed.rst

```diff
@@ -0,0 +1,480 @@
+    # Stop after 12 hours of training or when reaching 10 epochs (string)
+    trainer = Trainer(max_time="00:12:00:00", max_epochs=10)
+
+    # Stop after 1 day and 5 hours (dict)
+    trainer = Trainer(max_time={"days": 1, "hours": 5})
+
+Learn more in our :ref:`trainer_flags` guide.
+
+
+----------------
+
+****************************
+Control Validation Frequency
+****************************
+
+Check validation every n epochs
+===============================
+
+**Use when:** You have a small dataset, and want to run less validation checks.
+
+You can limit validation check to only run every n epochs using the `check_val_every_n_epoch` Trainer flag.
+
+.. testcode::
+
+    # DEFAULT
+    trainer = Trainer(check_val_every_n_epoch=1)
+
+
+Set validation check frequency within 1 training epoch
+======================================================
+
+**Use when:** You have a large dataset, and want to run mid-epoch validation checks.
+
+For large datasets, it's often desirable to check validation multiple times within a training loop.
+Pass in a float to check that often within 1 training epoch. Pass in an int `k` to check every `k` training batches.
+Must use an `int` if using an `IterableDataset`.
+
+.. testcode::
+
+    # DEFAULT
+    trainer = Trainer(val_check_interval=0.95)
+
+    # check every .25 of an epoch
+    trainer = Trainer(val_check_interval=0.25)
+
+    # check every 100 train batches (ie: for `IterableDatasets` or fixed frequency)
+    trainer = Trainer(val_check_interval=100)
+
+Learn more in our :ref:`trainer_flags` guide.
+
+----------------
+
+******************
+Limit Dataset Size
+******************
+
+Use data subset for training, validation, and test
+==================================================
+
+**Use when:** Debugging or running huge datasets.
+
+If you don't want to check 100% of the training/validation/test set set these flags:
+
+.. testcode::
+
+    # DEFAULT
+    trainer = Trainer(
+        limit_train_batches=1.0,
+        limit_val_batches=1.0,
+        limit_test_batches=1.0
+    )
+
+    # check 10%, 20%, 30% only, respectively for training, validation and test set
+    trainer = Trainer(
+        limit_train_batches=0.1,
+        limit_val_batches=0.2,
+        limit_test_batches=0.3
+    )
+
+If you also pass ``shuffle=True`` to the dataloader, a different random subset of your dataset will be used for each epoch; otherwise the same subset will be used for all epochs.
+
+.. note:: ``limit_train_batches``, ``limit_val_batches`` and ``limit_test_batches`` will be overwritten by ``overfit_batches`` if ``overfit_batches`` > 0. ``limit_val_batches`` will be ignored if ``fast_dev_run=True``.
+
+.. note:: If you set ``limit_val_batches=0``, validation will be disabled.
+
+Learn more in our :ref:`trainer_flags` guide.
+
+-----
+
+*********************
+Preload Data Into RAM
+*********************
+
+When your training or preprocessing requires many operations to be performed on entire dataset(s), it can
```

Comment:
```suggestion
*********************
Preload Data Into RAM
*********************

**Use when:** you need access to all samples in a dataset at once. 

When your training or preprocessing requires many operations to be performed on entire dataset(s), it can
```

adding a "use when" idea here :) 
(I wrote these docs)